java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TraceScope.scala

```diff
@@ -47,67 +47,68 @@ private[java] object TraceScope {
     case object Noop extends Scope
   }
 
-  def fromIOLocal[F[_]: LiftIO: Sync](
+  private[java] def fromStateful[F[_]: Sync: Stateful[*[_], Scope]](
```

Comment:
Unlike `Local`, it's not as obvious what other instances we're bringing to the table here.  `StateT` over a `Concurrent` is generally a bad idea.  `StateT` over a `MonadError` can go up in smoke.  

Could something clever be done with a `Ref`?  

Can we interoperate with synchronous Java with a *gulp* `ThreadLocal`?  Uh... probably not, because we also need a Sync?